MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -0,0 +1,275 @@
+import Polyline
+import MapboxDirections
+import AVFoundation
+import MapboxMobileEvents
+
+let SecondsBeforeCollectionAfterFeedbackEvent: TimeInterval = 20
+
+struct EventDetails {
+    var originalRequestIdentifier: String?
+    var requestIdentifier: String?
+    var coordinate: CLLocationCoordinate2D?
+    var originalGeometry: Polyline?
+    var originalDistance: CLLocationDistance?
+    var originalEstimatedDuration: TimeInterval?
+    var originalStepCount: Int?
+    var geometry: Polyline?
+    var distance: CLLocationDistance?
+    var estimatedDuration: TimeInterval?
+    var stepCount: Int?
+    var created: Date
+    var startTimestamp: String?
+    var platform: String
+    var operatingSystem: String
+    var device: String
+    var sdkIdentifier: String
+    var sdkVersion: String
+    var eventVersion: Int
+    var profile: String
+    var simulation: Bool
+    var sessionIdentifier: String
+    var distanceCompleted: CLLocationDistance
+    var distanceRemaining: TimeInterval
+    var durationRemaining: TimeInterval
+    var rerouteCount: Int
+    var volumeLevel: Int
+    var screenBrightness: Int
+    var batteryPluggedIn: Bool
+    var batteryLevel: Float
+    var applicationState: String
+    
+    init(routeController: RouteController, session: SessionState) {
+        created = Date()
+        if let start = session.departureTimestamp?.ISO8601 {
+            startTimestamp =  start
+        }
+        
+        platform = ProcessInfo.systemName
+        operatingSystem = "\(ProcessInfo.systemName) \(ProcessInfo.systemVersion)"
+        device = UIDevice.current.machine
+        
+        sdkIdentifier = routeController.usesDefaultUserInterface ? "mapbox-navigation-ui-ios" : "mapbox-navigation-ios"
+        sdkVersion = String(describing: Bundle(for: RouteController.self).object(forInfoDictionaryKey: "CFBundleShortVersionString")!)
+        
+        eventVersion = 2
```

Comment:
All the events produced by a particular version of this library should have the same version, right? Let’s make sure that’s the case by pulling this value out into a constant. No need for a property on the struct; just use the constant in `convertedToDictionary()`.